spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -457,6 +459,13 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
         PropertyUtil.propertyAsBoolean(
             options, CatalogProperties.CACHE_ENABLED, CatalogProperties.CACHE_ENABLED_DEFAULT);
 
+    SparkSession sparkSession = SparkSession.active();
+    boolean cacheCaseSensitive =
+        PropertyUtil.propertyAsBoolean(
+            options,
+            CatalogProperties.CACHE_CASE_SENSITIVE,
+            Boolean.parseBoolean(sparkSession.conf().get("spark.sql.caseSensitive")));
```

Comment:
Just realized that even the util method uses the constant, I would have thought it would use the SqlConf object val, oh well.  

It's minor but could we can add that to SparkUtil if we are anyway in Spark3.1 code, if its quick?

Reply:
I could add the `caseSensitive(SparkSession)` helper method to `SparkUtil` in 3.1 as well, but where do I stop? Do I then go on to rewrite the other places in 3.1 that should then use this helper instead? I didn't do so because this would touch more files unrelated to this change. @szehon-ho let me know what you prefer.

Reply:
Yea, i would say if its just a few lines, add the method to SparkUtil and then use it here, so at least this code is same across three versions.

Reply:
Ok, done.